charts/pvault-server/templates/deployment.yaml

```diff
@@ -94,78 +108,72 @@ spec:
             successThreshold: {{ .Values.startupProbe.successThreshold }}
             failureThreshold: {{ .Values.startupProbe.failureThreshold }}
           {{- end }}
-          resources:
-            {{- toYaml .Values.resources | nindent 12 }}
-          {{- if or .Values.pvault.log.datadogAPMEnable .Values.dbCARootCertificate .Values.pvault.directEnvVars  }}
+          resources: {{- toYaml .Values.resources | nindent 12 }}
           env:
-          {{- if .Values.pvault.log.datadogAPMEnable }}
+            {{- /* If CPU and memory limits are not specified for a container, and you use the downward API to try to expose that information, then the kubelet defaults to exposing the maximum allocatable value for CPU and memory based on the node allocatable calculation. */}}
+            - name: GOMAXPROCS
+              valueFrom:
+                resourceFieldRef:
+                  resource: {{ ternary "limits.cpu" "requests.cpu" (empty .Values.resources.requests.cpu) }}
+            - name: GOMEMLIMIT
+              valueFrom:
+                resourceFieldRef:
+                  resource: {{ ternary "limits.memory" "requests.memory" (empty .Values.resources.requests.memory) }}
```

Comment:
Why are we doing that? Also, is that based on https://www.riverphillips.dev/blog/go-cfs/

Reply:
It reduces resource contention on the node by signaling Golang how many cores/memory it should use.

This blog also explains why it helps: https://metoro.io/blog/go-production-performance-gotcha-gomaxprocs

Reply:
Updated `GOMEMLIMIT` to set `limits.memory` only, we can revisit later with more data.

Reply:
If the limits.memory is not set, then the value for GOMEMLIMIT will be based on approximation? wouldn't it be preferable to net set it at all?

Reply:
If not set, the raw node capacity is used which is higher than the allocatable memory reported by kubelet (it reserves memory for kernel + kubelet operation).